### DISCUSSION TITLE
Feasibility of using ELRS as Remote ID beacon

### DISCUSSION THREAD

**Discussion by CBRrunner** [ExpressLRS/ExpressLRS#1934] [Ideas]
With remote ID coming to the US in the next year what is the feasibility of using the WIFI or Bluetooth on the ELRS RX, on GPS equipped drones to transmit the data required to fulfill the FAA regulations? 

The modules I have seen are over $100 and weigh way too much to really be viable for light weight drones. IF ELRS could do it all we could have a basically free solution to one of the most intrusive regulations to hit the hobby that I can remember. 

I would think a flight controller with a GPS and ELRS module would have all the data telemetry necessary fulfill the FAA stuff. I guess the only hurdle would be approval, if there is even a process for that. I'm still looking into if it just needs to meet the requirements of the reg and we are good to go. 

I have no idea how to code but more than willing to support in any way that I can.

**Comment by CBRrunner** [comment]
ok the requirements of Federal Aviation Administration 14 CFR Parts 1, 11, 47, 48, 89, 91, and 107 

89.320 Minimum performance requirements for remote identification broadcast modules. 

Remote identification broadcast modules must meet the following minimum performance 
requirements:
(a) Take-off location. The remote identification broadcast module must be capable of determining the take-off location of the unmanned aircraft. 
(b) Time mark. The time mark message element must be synchronized with all other remote identification message elements.
(c) Self-Testing and monitoring. (1) Prior to take-off, the remote identification broadcast module must automatically test the remote identification functionality and notify the person manipulating the flight controls of the unmanned aircraft system of the result of the test.
(2) The remote identification broadcast module must continuously monitor the remote identification functionality from takeoff to shutdown and must provide notification of malfunction or failure to the person manipulating the flight controls of the unmanned aircraft 
system.

447
(d) Tamper resistance. The remote identification broadcast module must be designed and produced in a way that reduces the ability of a person to tamper with the remote identification functionality. 
(e) Error correction. The remote identification broadcast module must incorporate error correction in the broadcast of the message elements in § 89.315.
(f) Interference considerations. The remote identification broadcast module must not interfere with other systems or equipment installed on compatible unmanned aircraft, and other systems or equipment installed on compatible unmanned aircraft must not interfere with the 
remote identification equipment.

(g) Message broadcast. 
(1) The remote identification broadcast module must be capable of broadcasting the message elements in § 89.315 using a non-proprietary broadcast specification and using radio frequency spectrum compatible with personal wireless devices in accordance 
with part 15 of title 47, Code of Federal Regulations, where operations may occur without an FCC individual license. 
(2) The remote identification broadcast module must be designed to maximize the range at which the broadcast can be received, while complying with 47 CFR part 15 and any other applicable laws in effect as of the date the declaration of compliance is submitted to the FAA for acceptance.

(h) Message elements performance requirements. (1) The reported geometric position of the unmanned aircraft must be accurate to within 100 feet of the true position, with 95 percent probability.
(2) The reported geometric altitude of the unmanned aircraft must be accurate to within 150 feet of the true geometric altitude, with 95 percent probability.

448
(3) The reported geometric position of the take-off location must be accurate to within 100 feet of the true geometric position, with 95 percent probability.
(4) The reported geometric altitude of the take-off location must be accurate to within 150 feet of the true geometric altitude, with 95 percent probability.
(5) The remote identification broadcast module must broadcast the latitude, longitude, and geometric altitude of the unmanned aircraft no later than 1.0 seconds from the time of measurement to the time of broadcast. 
(6) The remote identification broadcast module must broadcast the message elements at a rate of at least 1 message per second.

**Comment by CBRrunner** [comment]
And acceptance:

 89.405 Submission of a means of compliance for FAA acceptance.
(a) Eligibility. Any person may submit a means of compliance for acceptance by the FAA.

449
(b) Required information. A person requesting acceptance of a means of compliance must submit the following information to the FAA in a form and manner acceptable to the Administrator:

(1) The name of the person or entity submitting the means of compliance, the name of the main point of contact for communications with the FAA, the physical address, email address, and other contact information.
(2) A detailed description of the means of compliance.
(3) An explanation of how the means of compliance addresses all of the minimum performance requirements established in subpart D of this part so that any standard remote identification unmanned aircraft or remote identification broadcast module designed and produced in accordance with such means of compliance meets the remote identification requirements of this part.
(4) Any substantiating material the person wishes the FAA to consider as part of the request.
(c) Testing and validation. A means of compliance submitted for acceptance by the FAA must include testing and validation procedures for persons responsible for the production of standard remote identification unmanned aircraft or remote identification broadcast modules to demonstrate through analysis, ground test, or flight test, as appropriate, how the standard remote identification unmanned aircraft or the remote identification broadcast module performs its intended functions and meets the requirements in subpart D of this part, including any applicable FAA performance requirements for radio station operation.

450
§ 89.410 Acceptance of a means of compliance.
(a) A person requesting acceptance of a means of compliance must demonstrate to the Administrator that the means of compliance addresses all of the requirements of subparts D and E of this part, and that any standard remote identification unmanned aircraft or remote identification broadcast module designed and produced in accordance with such means of compliance would meet the performance requirements of subpart D of this part.
(b) The Administrator will evaluate a means of compliance that is submitted to the FAA and may request additional information or documentation, as needed, to supplement the submission. 
(c) If the Administrator determines the person has demonstrated that the means of compliance meets the requirements of subparts D and E, the FAA will notify the person that the Administrator has accepted the means of compliance.

**Comment by CBRrunner** [comment]
This is a little module from the EU that just takes telemetry data and transmits it over BlueTooth. But they want $150 for it. 

https://www.aerobits.pl/product/idme/

I feel this would make a compelling case to explore the functionality for ELRS.

**Comment by WZ9V** [comment]
It would be great if the broadcast portion of Remote ID could be accomplished using the GPS telemetry data coming from a GPS enabled FC.  I would love an OpenSource solution like that if possible.

**Comment by matthew73210** [comment]
Hey all,
Not sure if it's okay to jump in here. But in France devices over 800g must transmit telemetry data (device name, owner, GPS etc...) but it can be done over a simple wifi telegram (not even sure if it needs a hotspot). A guy on a forum [Balise wifi](https://forum.wearefpv.fr/topic/13331-tuto-balise-wifi-de-signalement-%C3%A9lectronique-drone-800g-legit-fr/) used a simple esp to do all the work. As ELRS can poll data from the controller and already sends data back, this could be an easy way to get the regs in line. And make ELRS even more open to the whole community.
I wont go into the right and wrong of it, it's how the law is written in France.

**Reply by peters294** [reply]
It seems that the regulations in the EU are less restrictive than those of the FAA. Do you know anything about this topic?
